crypto/src/main/scala/org/bitcoins/crypto/CryptoRuntime.scala

```diff
@@ -213,16 +208,50 @@ trait CryptoRuntime {
     sum.bytes
   }
 
+  /** Adds two SecpPoints together and correctly handles the point at infinity (0x00). */
+  def add(point1: SecpPoint, point2: SecpPoint): SecpPoint = {
+    (point1, point2) match {
+      case (SecpPointInfinity, p) => p
+      case (p, SecpPointInfinity) => p
+      case (p1: SecpPointFinite, p2: SecpPointFinite) =>
+        val pk1 = p1.toPublicKey
+        val pk2 = p2.toPublicKey
+
+        if (
+          (pk1.bytes.head ^ pk2.bytes.head) == 0x01 && pk1.bytes.tail == pk2.bytes.tail
```

Comment:
Is `.head` safe here? I don't see invariants in `SecpPoint` that prevents empty byte vectors

Reply:
`ECPublicKey` has the `isFullyValid` invariant that guarantees that `pk.bytes` returns a valid compressed pubkey (beginning in either `02` or `03`)